ArtOfIllusion/src/artofillusion/view/ViewerScaleControl.java

```diff
@@ -88,4 +96,34 @@ public String getName()
   {
     return Translate.text("Magnification");
   }
+  
+  /*
+     For reasons unexplained to me, 'view.isPerspective()' appears always "true",
+     inside a ValueChecker class even if 'view' is set public. In ViewerCanvas the value
+     is correct though. Hence the two checkers.
+  */
+
+  private class ParallelChecker implements ValueChecker
+  {
+     ParallelChecker()
+     {}
+
+     @Override
+     public boolean isValid(double value)
+     { 
+       return view.getCamera().getDistToScreen()*100/value >= ViewerCanvas.MINIMUM_ZOOM_DISTANCE;
+     }
+  }
+
+  private class PerspectiveChecker implements ValueChecker
+  {
+     PerspectiveChecker()
+     {}
+
+     @Override
+     public boolean isValid(double value)
+     {
+       return value >= ViewerCanvas.MINIMUM_ZOOM_DISTANCE;
+     }
+  }
```

Comment:
Could you show me what the "Checker" code looked like when it was misbehaving? I think that you are running into some inner/outer class fine points. This *should* work with just a single checker.

To properly use OuterClass variables, the InnerClass constructor must be something like `zoomChecker(ViewerScaleControl control)` and then line 41 or so would be:

`scaleField.setValueChecker(new zoomChecker(this));`

Then you would call `control.view.isPerspective()` inside the checker.

Reply:
It would have looked, basically what I just pushed but without arguments in the constructor and the view of the `ViewerScaleControl` set public (or default) -- (not the `view` inside the widget). Probably had `else` there but I removed that. 

I only set the ViewerCanvas to the checker, that's all that it needs.

Reply:
Thanks. :)

Reply:
looks much better now!